components/sock_utils/include/netdb_macros.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#pragma once
+
+#ifndef NI_NUMERICHOST
```

Comment:
Minor: Can the 'NI_MAXHOST', 'NI_MAXSERV' be added here to specify the maximum sizes required for the buffers?
This will allow to avoid magic numbers.

Reply:
will use these number when allocating buffers in tests

Reply:
used `NI_MAXSERV` for buffer size in tests

Reply:
unfortunately I had to revert this change to pass the CI.
This is defined in newlib headers in IDF, and newlib is not supported on linux target.
I can use some system header (which works on my machine), but wouldn't work on build machine.